packages/widgetbook/lib/src/navigation/router.dart

```diff
@@ -51,13 +52,15 @@ bool _parseBoolQueryParameter({
 }
 
 GoRouter createRouter({
-  required PreviewProvider previewProvider,
+  required UseCasesProvider useCasesProvider,
 }) {
   final router = GoRouter(
     redirect: (context, routerState) {
       final path = routerState.queryParams['path'];
 
-      previewProvider.selectUseCaseByPath(path);
+      if (path != null) {
+        useCasesProvider.selectUseCaseByPath(path);
+      }
```

Comment:
In the case of a proper `go_router` implementation, I think this should not be necessary. Let me know once you're available so we can talk about this.